<?xml version="1.0" encoding="UTF-8"?><!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml" lang="en"><head><meta http-equiv="Content-Type" content="text/html;charset=UTF-8"/><link rel="stylesheet" href="../jacoco-resources/report.css" type="text/css"/><link rel="shortcut icon" href="../jacoco-resources/report.gif" type="image/gif"/><title>GlobalExceptionHandler.java</title><link rel="stylesheet" href="../jacoco-resources/prettify.css" type="text/css"/><script type="text/javascript" src="../jacoco-resources/prettify.js"></script></head><body onload="window['PR_TAB_WIDTH']=4;prettyPrint()"><div class="breadcrumb" id="breadcrumb"><span class="info"><a href="../jacoco-sessions.html" class="el_session">Sessions</a></span><a href="../index.html" class="el_report">user-management-platform</a> &gt; <a href="index.source.html" class="el_package">es.ibm.usermanagement.exception.handler</a> &gt; <span class="el_source">GlobalExceptionHandler.java</span></div><h1>GlobalExceptionHandler.java</h1><pre class="source lang-java linenums">package es.ibm.usermanagement.exception.handler;


import es.ibm.usermanagement.dto.ErrorResponse;
import es.ibm.usermanagement.exception.custom.AsyncTaskFailureException;
import es.ibm.usermanagement.exception.custom.InvalidParamException;
import es.ibm.usermanagement.exception.custom.UserAlreadyExistsException;
import es.ibm.usermanagement.exception.custom.UserNotFoundExeption;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.MethodArgumentNotValidException;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.RestControllerAdvice;
import lombok.extern.slf4j.Slf4j;

import java.time.LocalDateTime;

<span class="nc" id="L18">@Slf4j</span>
@RestControllerAdvice
<span class="nc" id="L20">public class GlobalExceptionHandler {</span>

    @ExceptionHandler(UserAlreadyExistsException.class)
    public ResponseEntity&lt;ErrorResponse&gt; handleUserExists(UserAlreadyExistsException ex) {
<span class="nc" id="L24">        log.error(String.format(&quot;Enter in %s handler&quot;,ex.getClass()));</span>

<span class="nc" id="L26">        ErrorResponse error = new ErrorResponse(</span>
<span class="nc" id="L27">                LocalDateTime.now(),</span>
<span class="nc" id="L28">                HttpStatus.CONFLICT.value(),</span>
<span class="nc" id="L29">                ex.getMessage()</span>
        );
<span class="nc" id="L31">        return new ResponseEntity&lt;&gt;(error, HttpStatus.CONFLICT);</span>
    }


    @ExceptionHandler(AsyncTaskFailureException.class)
    public ResponseEntity&lt;ErrorResponse&gt; handleAsyncTaskFailureException(AsyncTaskFailureException ex) {
<span class="nc" id="L37">        log.error(String.format(&quot;Enter in %s handler&quot;,ex.getClass()));</span>
<span class="nc" id="L38">        ErrorResponse error = new ErrorResponse(</span>
<span class="nc" id="L39">                LocalDateTime.now(),</span>
<span class="nc" id="L40">                HttpStatus.INTERNAL_SERVER_ERROR.value(),</span>
<span class="nc" id="L41">                ex.getMessage()</span>
        );
<span class="nc" id="L43">        return new ResponseEntity&lt;&gt;(error, HttpStatus.INTERNAL_SERVER_ERROR);</span>
    }

    @ExceptionHandler(UserNotFoundExeption.class)
    public ResponseEntity&lt;ErrorResponse&gt; handleUserNotFoundExeption(UserNotFoundExeption ex) {
<span class="nc" id="L48">       log.error(String.format(&quot;Enter in %s handler&quot;,ex.getClass()));</span>
<span class="nc" id="L49">        ErrorResponse error = new ErrorResponse(</span>
<span class="nc" id="L50">                LocalDateTime.now(),</span>
<span class="nc" id="L51">                HttpStatus.NOT_FOUND.value(),</span>
<span class="nc" id="L52">                ex.getMessage()</span>
        );
<span class="nc" id="L54">        return new ResponseEntity&lt;&gt;(error, HttpStatus.NOT_FOUND);</span>
    }


    @ExceptionHandler(InvalidParamException.class)
    public ResponseEntity&lt;ErrorResponse&gt; handleInvalidParamException(InvalidParamException ex) {
<span class="nc" id="L60">        log.error(String.format(&quot;Enter in %s handler&quot;,ex.getClass()));</span>
<span class="nc" id="L61">        ErrorResponse error = new ErrorResponse(</span>
<span class="nc" id="L62">                LocalDateTime.now(),</span>
<span class="nc" id="L63">                HttpStatus.BAD_REQUEST.value(),</span>
<span class="nc" id="L64">                String.format(&quot;Invalid %s&quot;, ex.getParameter())</span>
        );
<span class="nc" id="L66">        return new ResponseEntity&lt;&gt;(error, HttpStatus.BAD_REQUEST);</span>
    }

    @ExceptionHandler(MethodArgumentNotValidException.class)
    public ResponseEntity&lt;ErrorResponse&gt; handleValidationException(MethodArgumentNotValidException ex) {
<span class="nc" id="L71">        log.error(String.format(&quot;Enter in %s handler&quot;,ex.getClass()));</span>
<span class="nc" id="L72">        ErrorResponse error = new ErrorResponse(</span>
<span class="nc" id="L73">                LocalDateTime.now(),</span>
<span class="nc" id="L74">                HttpStatus.BAD_REQUEST.value(),</span>
<span class="nc" id="L75">                String.format(&quot;Invalid %s&quot;, ex.getBindingResult().getFieldErrors().get(0).getField())</span>
        );
<span class="nc" id="L77">        return new ResponseEntity&lt;&gt;(error, HttpStatus.BAD_REQUEST);</span>
    }

}
</pre><div class="footer"><span class="right">Created with <a href="http://www.jacoco.org/jacoco">JaCoCo</a> 0.8.12.202403310830</span></div></body></html>